### PR TITLE
PN_9307

### PR DESCRIPTION
**Pull request by azhansultan**
*Added new steps and methods
*Added new feature file 015_033B_annullaeliminaIndirizzoPEC.feature

### CODE REVIEW

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 1]
```diff
@@ -347,8 +347,14 @@ public void clickSuEliminaPec() {
     public String waitLoadPopUpElimina() {
         By titlePopUp = By.id("dialog-title");
         By subTitlePopUp = By.id("dialog-description");
+        By cancelButton = By.id("buttonAnnulla");
+        By confirmButton = By.id("buttonConferma");
         this.getWebDriverWait(10).withMessage("Non è stato caricato il titolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(titlePopUp));
         this.getWebDriverWait(10).withMessage("Non è stato caricato il sottotitolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(subTitlePopUp));
+        this.getWebDriverWait(10).withMessage("Non è stato caricato il sottotitolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(cancelButton));
+        this.getWebDriverWait(10).withMessage("Non è stato caricato il titolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(confirmButton));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: replace the word `pop-up` with `modal`

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 2]
```diff
@@ -358,6 +364,12 @@ public void clickSuConfermaElimina() {
         this.element(confermaRimuoviPECBy).click();
     }
 
+    public void clickSuAnnullaElimina() {
+        By annullaRimuoviPECBy = By.id("buttonAnnulla");
+        this.getWebDriverWait(30).withMessage("Non è stato possibile cliccare sul bottone annulla").until(ExpectedConditions.elementToBeClickable(annullaRimuoviPECBy));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: preferably use **10 seconds**

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 3]
```diff
@@ -358,6 +364,12 @@ public void clickSuConfermaElimina() {
         this.element(confermaRimuoviPECBy).click();
     }
 
+    public void clickSuAnnullaElimina() {
+        By annullaRimuoviPECBy = By.id("buttonAnnulla");
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: The element is already declared at the top of the file:
https://github.com/pagopa/pn-frontend-e2e/blob/133de6c62d0b35c4ef53e27c90ae3bd486431217/src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java#L65-L66

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java (hunk 1 above)
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: Did you check if this update doesn't break any other test case?

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 4]
```diff
@@ -369,6 +381,17 @@ public boolean siControllaNonPresenzaPEC() {
         }
     }
 
+    public boolean siControllaPresenzaPEC() {
+        try {
+            By pecField = By.id("legalContacts");
+            this.getWebDriverWait(10).until(ExpectedConditions.visibilityOfElementLocated(pecField));
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: The element is already declared at the top of the file:
https://github.com/pagopa/pn-frontend-e2e/blob/133de6c62d0b35c4ef53e27c90ae3bd486431217/src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java#L71-L72

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 5]
```diff
@@ -686,7 +686,20 @@ public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneConferma() {
...
+    public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneAnnulla() {
+        logger.info("Si clicca sul bottone conferma");
+        if (recapitiDestinatarioPage.waitLoadPopUpElimina().equalsIgnoreCase("Rimuovi PEC")) {
+            recapitiDestinatarioPage.clickSuAnnullaElimina();
+        } else {
+            recapitiDestinatarioPage.clickSuChiudiPopUp();
+            recapitiDestinatarioPage.eliminaNuovaPec();
+            recapitiDestinatarioPage.eliminaPecEsistente();
+        }
+    }
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]: I think there is already a step or method that should do this kind of action

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 6]
```diff
@@ -698,6 +711,20 @@ public void nellaPaginaITuoiRecapitiSiControllaCheLIndirizzoPecNonSiaPresente()
...
+    public void nellaPaginaITuoiRecapitiSiControllaCheLIndirizzoPecPresente() {
+        logger.info("Si controlla che la PEC non stata eliminata");
+
+        if (recapitiDestinatarioPage.siControllaPresenzaPEC()) {
+            logger.info("La PEC non è stata eliminata");
+        } else {
+            logger.error("La PEC è stata eliminata");
+            Assert.fail("La PEC è stata eliminata");
+        }
+    }
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:724]: Same behaviour of some method above, check if you can optimize this process and use the global webelement at the top of the file

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java (hunk 1 above)
Comment (azhansultan) [reply src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: no , but every pop-up while deleting email has annulla and conderma button.

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 5 above)
Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]: yes there is similar method , but it doesnt verify that pop-up contains "Rimuovi PEC", and doest have a logic to implement if pop-up is not "Rimuovi PEC"

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 6 above)
Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:724]: done

File: src/test/resources/feature/3-destinatario/personaFisica/altriRecapiti/015_033_eliminaIndirizzoPEC.feature [hunk 7]
```diff
@@ -5,7 +5,7 @@ Feature: la persona fisica elimina l'indirizzo pec
   @TA_eliminaPECPF
   @recapitiPF
 
-  Scenario: PN-9307 - la persona fisica elimina l'indirizzo pec
+  Scenario: PN-9307 -A33 la persona fisica elimina l'indirizzo pec
```
Comment (afus0) [inline src/test/resources/feature/3-destinatario/personaFisica/altriRecapiti/015_033_eliminaIndirizzoPEC.feature]:
```suggestion
  Scenario: PN-9307-A33 - La persona fisica elimina l'indirizzo pec
```

File: src/test/resources/feature/3-destinatario/personaFisica/altriRecapiti/015_033B_annullaeliminaIndirizzoPEC.feature [hunk 8]
```diff
@@ -0,0 +1,16 @@
+Feature: la persona fisica elimina l'indirizzo pec
+
+  @TestSuite
+  @PF
+  @TA_annulaEliminaPECPF
+  @recapitiPF
+
+  Scenario: PN-9307 -B33- la persona fisica elimina l'indirizzo pec
```
Comment (afus0) [inline src/test/resources/feature/3-destinatario/personaFisica/altriRecapiti/015_033B_annullaeliminaIndirizzoPEC.feature]:
```suggestion
  Scenario: PN-9307-B33 - La persona fisica elimina l'indirizzo pec
```

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 9]
```diff
@@ -347,8 +347,14 @@ public void clickSuEliminaPec() {
     public String waitLoadPopUpElimina() {
         By titlePopUp = By.id("dialog-title");
         By subTitlePopUp = By.id("dialog-description");
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il titolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(titlePopUp));
-        this.getWebDriverWait(10).withMessage("Non è stato caricato il sottotitolo del pop-up").until(ExpectedConditions.visibilityOfElementLocated(subTitlePopUp));
+        By cancelButton = By.id("buttonAnnulla");
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: This one is already present, use it and do the same for the confirmButton
https://github.com/pagopa/pn-frontend-e2e/blob/10524309fbd27d6c7d78103d86d876080fe07053/src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java#L65-L66

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 10]
```diff
@@ -347,8 +347,14 @@ public void clickSuEliminaPec() {
...
+        By confirmButton = By.id("buttonConferma");
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: Do the same as the cancel one

File: src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java [hunk 11]
```diff
@@ -358,9 +364,14 @@ public void clickSuConfermaElimina() {
         this.element(confermaRimuoviPECBy).click();
     }
 
-    public boolean siControllaNonPresenzaPEC() {
+    public void clickSuAnnullaElimina() {
+        this.getWebDriverWait(10).withMessage("Non è stato possibile cliccare sul bottone annulla").until(ExpectedConditions.elementToBeClickable(buttonAnnullaEliminazioneInPopUp));
+        buttonAnnullaEliminazioneInPopUp.click();
+    }
```
Comment (afus0) [inline src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java]: Already exist: 
https://github.com/pagopa/pn-frontend-e2e/blob/10524309fbd27d6c7d78103d86d876080fe07053/src/main/java/it/pn/frontend/e2e/common/RecapitiDestinatarioPage.java#L564-L566

Just add the same check on the existent one

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java [hunk 12]
```diff
@@ -686,18 +686,45 @@ public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneConferma() {
...
+    public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneAnnulla() {
+        logger.info("Si clicca sul bottone annnulla");
+        if (recapitiDestinatarioPage.waitLoadPopUpElimina().equalsIgnoreCase("Rimuovi PEC")) {
+            recapitiDestinatarioPage.clickSuAnnullaElimina();
+        } else {
+            recapitiDestinatarioPage.clickSuChiudiPopUp();
+            recapitiDestinatarioPage.eliminaNuovaPec();
+            recapitiDestinatarioPage.eliminaPecEsistente();
+        }
+    }
```
Comment (afus0) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java]: This one already exist:
https://github.com/pagopa/pn-frontend-e2e/blob/1f3209a14ec44180d6b619fe22f81eb56ce131e9/src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaGiuridica/RecapitiPGPagoPaTest.java#L99-L103

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 6 above)
Comment (afus0) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:724]: This one is already present here:
https://github.com/pagopa/pn-frontend-e2e/blob/10524309fbd27d6c7d78103d86d876080fe07053/src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java#L702-L712

**Review by afus0 — CHANGES_REQUESTED** [review]

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java (hunk 6 above)
Comment (azhansultan) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java:724]: it is two different steps with, one is checking if pec eleminated, the other one checks that not deleted. I already did one optimization in siControllaPresenzaPec method

**Review by afus0 — APPROVED** [review]

**Pull request merged**